lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -373,6 +374,112 @@
         default: no
         version_added: "2.8"
 
+    entrust_cert_type:
+        description:
+            - The type of certificate product to request.
+        type: str
+        default: STANDARD_SSL
+        choices: [ 'STANDARD_SSL', 'ADVANTAGE_SSL', 'UC_SSL', 'EV_SSL', 'WILDCARD_SSL', 'PRIVATE_SSL', 'PD_SSL', 'CODE_SIGNING', 'EV_CODE_SIGNING',
+        'CDS_INDIVIDUAL', 'CDS_GROUP', 'CDS_ENT_LITE', 'CDS_ENT_PRO', 'SMIME_ENT' ]
+        version_added: "2.9"
+
+    entrust_requester_email:
+        description:
+            - The email of the requester of the certificate (for tracking purposes).
+        type: str
+        version_added: "2.9"
+
+    entrust_requester_name:
+        description:
+            - The name of the requester of the certificate (for tracking purposes).
+        type: str
+        version_added: "2.9"
+
+    entrust_requester_phone:
+        description:
+            - The phone number of the requester of the certificate (for tracking purposes).
+        type: str
+        version_added: "2.9"
+
+    entrust_api_user:
+        description:
+            - The username for authentication to the Entrust ECS API.
+        type: str
+        version_added: "2.9"
+
+    entrust_api_key:
+        description:
+            - The key (password) for authentication to the Entrust ECS API.
+        type: str
+        version_added: "2.9"
+
+    entrust_api_client_cert_path:
+        description:
+            - The path of the client certificate used to authenticate to the Entrust ECS API.
+        type: path
+        version_added: "2.9"
+
+    entrust_api_client_cert_key_path:
+        description:
+            - The path of the key for the client certificate used to authenticate to the Entrust ECS API
+        type: path
+        version_added: "2.9"
+
+    entrust_reissue_if_expires_in:
+        description:
+            - If this value is set and an existing certificate is present, it will be reissued if it expires within this number of seconds.
+            - If not present or value is 0, it will be reissued if the certificate is already expired.
+            - For example, if you want to reissue a certificate if it is expiring within a month, set this to 2592000.
+              Certificates with more seconds remainingpast this number of seconds from when the module is called will be left as is.
+        type: int
+        version_added: "2.9"
```

Comment:
This also looks like something that should not be provider-specific, but available for all backends. IMO it should also be not part of this PR.

(Also, it would be nice if it is possible to use other units, like minutes, hours, and days.)

Reply:
My main concern with making "renew" generic is that it can be functionally different for different CAs. (e.g. whether they revoke the old cert, etc). I agree with separate pull request if added at all, and if I do make a PR for it I'll change it to use the same relative time paradigm as the other time fields.

Reply:
It depends a lot on what exactly "renew" is. If it just means "getting a new certificate for the same set of domains with the same other properties", no special magic would be required and this could work the same for all providers.

Right now this part could be done by first calling `openssl_certificate_info`, and then using the result to decide whether you want to renew or not.

If there are other steps, like revoking the old certificate, that should not be done by the same module IMO, for several reasons:
 1. it's against the Unix philosophy (do one thing, and do it well), and the module is already complex enough;
 2. users only want to revoke the old cert once they got the new cert and installed and activated it everywhere. Otherwise your site visitors suddenly get certificate errors if you revoke too early.

It's better to have a separate module for revocation. (For example, there's already a separate module for revoking certificates via ACME: https://docs.ansible.com/ansible/latest/modules/acme_certificate_revoke_module.html)

Reply:
We have a 'reissue' operation that revokes existing certificates, renew was chosen specifically because it doesn't, but I see what you mean. For now I'll leave it out, I think it complicates the module unnecessarily to be doing different things depending on the specific state and timing of the certificate that cert_path points to. Thanks for all your input! Appreciate the feedback.